### DISCUSSION TITLE
站点管理 栏目内站点卡片样式改进的个人建议

### DISCUSSION THREAD

**Discussion by hn315** [jxxghp/MoviePilot#3462] [讨论]
### 当前程序版本

v1.8.0

### 运行环境

Docker

### 功能改进类型

主程序

### 功能改进

      MP已非常好用了，谢谢开发者。最近我的PT站可能因为DNS污染的原因，老是似通非通。在MP 站点管理栏目内，总担心哪个站又不通了，站点卡片上不能直接看到联通的情况，于是总想去点一下“测试”或“浏览”，确定站点是否正常联通。PT站的联通与否与订阅、下载、统计、搜索、RSS等等许多功能关切。也许是强迫症，总想自己添加的PT站全部是正常联通的。有一个人建议：
      1、能不能在站点卡片右侧增加一个色彩颗粒柱状图，比如5个颗粒，绿色表示联通过，红色表示断联。
      2、鼠标移上去显示联通或断开的状态时间。
      3、这些通断的数据可以提取来自站点的订阅、RSS、数据统计是否正常等。
      4、一看站点卡片绿色柱状就证明站点联通正常，我感觉很实用直观。
      我不会画图，只能说说，下面贴个图表示样式。个人建议，仅供参考。
<img width="237" alt="捕获1" src="https://github.com/jxxghp/MoviePilot/assets/63461319/7d01066f-e0d1-4c75-9dc9-b5de90ae143c">

**Comment by jxxghp** [comment]
挺好，之前也想过，但每次测试都要访问站点的，定时会增加站点压力。

**Comment by hn315** [comment]
> 挺好，之前也想过，但每次测试都要访问站点的，定时会增加站点压力。

      谢谢J大亲自回复。是的，如果单独为这个显示状态去访问站点肯定是不合适，会增加站点压力和系统开销。

1、但是一般用MP的都会有订阅，订阅模式如果是RSS ，一般时间间隔都不长，每次后台会RSS访问订阅PT站，这个时候如果RSS成功返回数据就证明站点是联通的，写入数据库，点亮一颗绿色颗粒。如果订阅是搜索模式也会提取到是否PT站访问正常。
2、还有，大家一般会主动去搜索资源，能返回数据的也可以证明PT站联通成功。
3、还有一部分同学会用站点签到插件或其他各种插件，都会访问到PT站点，一样可提取状态数据。
4、MP里完全没有用的PT站，卡片就不显示联通状态，也可以直观知道哪些站没有用。

     我不知道表述清楚不。反正就是说不单为了这个卡片的显示而去主动测试PT站联通情况，而是利用MP在正常运行时，会有各种方式去连PT站的（当然也包括手动点卡片测试是否联通），这个时候只要把每一次连接不同PT站的状态提取出来存数据库，每站只保留最近10次（或5次）状态（包括联通与否、提取时间）。当用户点开站点管理时，供前台卡片UI提取数据显示。绿色柱状PT站卡片应该能增加站点管理的颜值。更能直观看到每个PT站的连通状态。实现这个颜值要时间去弄，我的单纯想法而已，不当之处请 J 大批评。
